src/shared/utilities/timeoutUtils.ts

```diff
@@ -113,6 +113,15 @@ export class Timeout {
 
         this._completed = true
     }
+
+    public reset(): void {
+        this._completed = false
+        this.timerPromise = new Promise<undefined>((resolve, reject) => {
+            this.timerReject = reject
+            this.timerResolve = resolve
+        })
+        this.refresh()
```

Comment:
Thinking about the goal here again, I don't understand why refresh() does not do what we want. 

1. The goal is to continue the timer if the "build" step in localLambdaRunner.ts finished without timing out. 
2. If it didn't timeout, refresh() should do what we want.
3. If it _did_ timeout during build, then... we should abort the run/debug task. Because the timeout was reached.

Reassigning `this.timerPromise` doesn't look right to me. It means that 

- anything that called [timer()](https://github.com/aws/aws-toolkit-vscode/blob/e70850f97618bec9d9b8a94f5877ffe62284145f/src/shared/utilities/timeoutUtils.ts#L83) will have a reference to the old timer, which is confusing and semantically incorrect.
- anything that called [waitTimeout()](https://github.com/aws/aws-toolkit-vscode/blob/e70850f97618bec9d9b8a94f5877ffe62284145f/src/shared/utilities/timeoutUtils.ts#L194) will still be waiting on the old timer. So reset() won't do what is expected.

We either need some way to reset the state of the _existing_  `timerPromise` or we need to reevaluate our requirements...

Reply:
> Thinking about the goal here again, I don't understand why refresh() does not do what we want.
> 
>     The goal is to continue the timer if the "build" step in localLambdaRunner.ts finished before timeout.
>     If it didn't timeout, refresh() should already do what we want (AFAICT).
>     If it did timeout during build, then... we should abort the run/debug task. Because the timeout was reached.

But what about builds that take longer than the timer, but still successfully build? I thought the idea of the timer was a way for the process to exit if it gets hung.

Reply:
>     Using this on a timer that has already called its callback will reactivate the timer.
> 
> That seems like exactly what we want, no?

Unfortunately, even when manually calling a refresh after a build that exceeds the timer, it doesn't appear to reactivate and the invoke step fails immediately. I'll look further into resetting the existing `timerPromise`.

Reply:
Conclusions from chat:

- Increase the default timeout to 90s instead of 30s. (If the timeout is exceeded then we should cancel the build. That is why we have the timeout. If builds are legitimately taking longer than we have allowed, we should increase the timeout default value.)
- Always call refresh() after the build completes, before the "attach" phase.